components/scoped-notification/index.jsx

```diff
@@ -0,0 +1,78 @@
+/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
+/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */
+
+// Implements the [Scoped Notification design pattern](https://lightningdesignsystem.com/components/scoped-notifications/) in React.
+// Based on SLDS v2.4.5
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import IconSettings from '../icon-settings';
+import Icon from '../icon';
+
+import { SCOPED_NOTIFICATION } from '../../utilities/constants';
+
+const propTypes = {
+	/**
+	 * CSS classes to be added to tag with `.slds-scoped-notification`. Uses `classNames` [API](https://github.com/JedWatson/classnames).
+	 */
+	className: PropTypes.oneOfType([
+		PropTypes.array,
+		PropTypes.object,
+		PropTypes.string,
+	]),
+	/**
+	 *  Theme for the scoped notification
+	 */
+	theme: PropTypes.oneOf(['dark', 'light']),
+	/**
+	 *  Icon for the scoped notification
+	 */
+	icon: PropTypes.string,
```

Comment:
Icon would the an `<Icon>` component. This should probably be `iconName` and folks should know that it's limited to the utility set of icons at this point. https://www.lightningdesignsystem.com/icons/#utility

Reply:
Ya, True. I went for a rather generic name, thanks for patching it up. I will take a note of it, and would update for the other prop proposals I have made elsewhere as well.